torch/csrc/Storage.cpp

```diff
@@ -164,14 +164,15 @@ static PyObject* THPStorage_pynew(
             allocator,
             /*resizable=*/true));
     THPObjectPtr item;
+
+    c10::Storage storage = *std::move(self->cdata);
```

Comment:
Why do you need to move the storage out of self temporarily 🤔

Reply:
If this is because of const correctness, this is misguided, for the same reason const Tensor& should be able to vend both mutable and immutable data ptrs